src/robot/libraries/Collections.py

```diff
@@ -851,13 +886,13 @@ def dictionary_should_not_contain_value(
 
     def dictionaries_should_be_equal(
         self,
-        dict1,
-        dict2,
-        msg=None,
-        values=True,
-        ignore_keys=None,
-        ignore_case=False,
-        ignore_value_order=False,
+        dict1: Mapping,
+        dict2: Mapping,
+        msg: "str | None" = None,
+        values: bool = True,
+        ignore_keys: "Sequence[Hashable] | str | None" = None,
```

Comment:
This could probably be `list[Hashable] | None` or `Sequence[Hashable] | None` if you prefer less strict type. Either way, list literals are converted to lists. Both also accept any sequence so there's no practical difference.